### DISCUSSION TITLE
Calling Streamlink from rtsp-simple-server on Windows10.

### DISCUSSION THREAD

**Discussion by rktr2009** [bluenviron/mediamtx#856] [Questions] [answered]
I am running rtsp-simple-server and publishing a stream to it with streamlink piped to ffmpeg and streamed to rtsp-simple-server on windows10.
Those commands are scripted in a .bat file which is registered as a background task with nssm. 
I am satisfied with it, but since it is running all the way, and to decrease the use of cpu, I am exploring the use of runOnDemand.

Whatever I do there though, it doesn't seem to work. 
Does runOnDemand work under windows?
Since I need streamlink for this specific stream, can I call an external command like streamlink or even a .bat file with streamlink scripted in there?

Server version is 0.17.17.

Thank you.

**Comment by aler9** [comment, marked as answer]
`runOnDemand` works fine on Windows, but it works with executables, and a .bat is not an executable; to launch it, you have to wrap it into `cmd`:

```yml
paths:
  mypath:
    source: publisher
    runOnDemand: cmd /k myscript.bat
```

**Reply by rktr2009** [reply]
@aler9 Thank you so much! It is now working with a .bat.

[a comment by an automated account is not shown]
